packager/hls/base/media_playlist.cc

```diff
@@ -181,7 +181,7 @@ class SegmentInfoEntry : public HlsEntry {
                    uint64_t segment_file_size,
                    uint64_t previous_segment_end_offset);
 
-  std::string ToString() override;
+  std::string ToString(std::string) override;
```

Comment:
Having to modify the signature of `ToString()` all over the place just to be able to do `entry->ToString("#EXT-X-SESSION-KEY")` doesn't feel right.

Can probably either add a method on `EncryptionInfoEntry` or a subclass, etc.

Reply:
@cosmin
I changed this to add a ToString(std::string) to `EncryptionInfoEntry` only and then cast entry to type EncryptionInfoEntry at L580-L581.